src/Metabor/Statemachine/Statemachine.php

```diff
@@ -237,4 +257,29 @@ public function getCurrentContext()
     {
         return $this->currentContext;
     }
+
+    /**
+     * @throws LockCanNotBeAcquiredException
+     */
+    protected function acquireLockOrThrowException()
+    {
+        if (!$this->acquireLock()) {
+            throw new LockCanNotBeAcquiredException('Lock can not be acquired!');
+        }
+    }
+
+    /**
+     * Use this function if you want to aquire lock before calling triggerEvent or checkTransitions.
+     * Lock is aquired automatically when calling dispatchEvent or checkTransitions.
+     *
+     * @return bool
+     */
+    public function acquireLock()
+    {
+        if ($this->mutex->isAcquired()) {
```

Comment:
or maybe just return $this->mutex->acquireLock(); as mutex already caches the lock execution